lib/pg_party/adapter_decorator.rb

```diff
@@ -63,49 +109,138 @@ def detach_partition(parent_table_name, child_table_name)
       PgParty.cache.clear!
     end
 
-    private
+    def partitions_for_table_name(table_name, include_subpartitions:, _accumulator: [])
+      select_values(%[
+          SELECT pg_inherits.inhrelid::regclass::text
+          FROM pg_tables
+          INNER JOIN pg_inherits
+            ON pg_tables.tablename::regclass = pg_inherits.inhparent::regclass
+          WHERE pg_tables.tablename = #{quote(table_name)}
+                    ]).each_with_object(_accumulator) do |partition, acc|
+        acc << partition
+        next unless include_subpartitions
+
+        partitions_for_table_name(partition, include_subpartitions: true, _accumulator: acc)
+      end
+    end
 
-    def create_partition(table_name, type, partition_key, **options)
-      modified_options = options.except(:id, :primary_key, :template)
-      template         = options.fetch(:template, true)
-      id               = options.fetch(:id, :bigserial)
-      primary_key      = options.fetch(:primary_key) { calculate_primary_key(table_name) }
+    def parent_for_table_name(table_name, traverse: false)
+      parent = select_values(%[
+          SELECT pg_inherits.inhparent::regclass::text
+          FROM pg_tables
+          INNER JOIN pg_inherits
+            ON pg_tables.tablename::regclass = pg_inherits.inhrelid::regclass
+          WHERE pg_tables.tablename = #{quote(table_name)}
```

Comment:
Did we want to add a clause to check `current_schema()` here as well?

Reply:
Done in https://github.com/webandtech/pg_party/commit/194486cafc08dba50b9e119513eebfa03105435e